contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,221 @@
+    Ignore FailurePolicyType = "Ignore"
+    Fail FailurePolicyType = "Fail"
+)
+
+type ExternalAdmissionHook struct {
+    // Operations is the list of operations this hook will be invoked on - Create, Update, or *
+    // for all operations. Defaults to '*'.
+    Operations []OperationType
+    // Resources are the resources this hook should be invoked on. '*' is all resources.
+    Resources []Resource
+    // Subresources is list of subresources. If non-empty, this hook should be invoked on 
+    // all combinations of Resources and Subresources. '*' is all subresources.
+    Subresources []string
+
+    // ClientConfig defines how to talk to the hook.
+    ClientConfig AdmissionHookClientConfig
+
+    // FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+    // allowed values are Ignore, Fail. Default value is Fail
+    FailurePolicy FailurePolicyType
+}
+
+type Resource struct {
+    // Group is the API group the resource belongs to.
+    Group string
+    // Resource is the name of the resource.
+    Resource string
+}
+
+type OperationType string
+
+const (
+    All OperationType = "*"
+    Create OperationType= "Create"
+    Update OperationType= "Update"
+)
+
+// AdmissionHookClientConfig contains the information to make a TLS
+// connection with the webhook
+type AdmissionHookClientConfig struct {
+    // Service is a reference to the service for this webhook. It must communicate
+	// on port 443
+	Service ServiceReference
+	// CABundle is a PEM encoded CA bundle which will be used to validate webhook's server certificate.
+	CABundle []byte
+}
+
+// ServiceReference holds a reference to Service.legacy.k8s.io
+type ServiceReference struct {
+	// Namespace is the namespace of the service
+	Namespace string
+	// Name is the name of the service
+	Name string
+}
+```
+
+## Synchronization of AdmissionControlConfiguration (**optional for alpha implement**)
+
+If the `initializer admission controller` and the `generic webhook admission
+controller` watch the `AdmissionControlConfiguration` and act upon deltas, their
+cached version of the configuration might be arbitrarily delayed. This makes it
+impossible to predict what initializer/hooks will be applied to newly created
+objects.
+
+We considered a few ways to make the behavior of the `initializer admission
+controller` and the `generic webhook admission controller` predictable.
+
+(I prefer #2. #1 is inefficient, #3 requires complex schema and is not intuitive)
+
+#### 1. Always do consistent read 
+
+The `initializer admission controller` and the `generic webhook admission
+controller` always do consistent read of the `AdmissionControlConfiguration`
+before applying the configuration to the incoming objects. This adds latency to
+every CREATE request. Because the two admission controllers are in the same
+process as the apiserver, the latency mainly consists of the consistent read
+latency of the backend storage (etcd), and the proto unmarshalling.
```

Comment:
@smarterclayton - i think I'm lost in what you're suggesting. Can you please clarify it?